zboxcore/sdk/read_counter_ctr.go

```diff
@@ -2,40 +2,43 @@ package sdk
 
 import (
 	"sync"
-
-	"github.com/0chain/gosdk/zboxcore/blockchain"
 )
 
 type blobberReadCounter struct {
 	m  map[string]int64
 	mu *sync.RWMutex
 }
 
-var brc blobberReadCounter
+var brc *blobberReadCounter
 
 func InitReadCounter() {
-	brc = blobberReadCounter{
-		m:  make(map[string]int64),
-		mu: &sync.RWMutex{},
+	if brc == nil {
+		brc = &blobberReadCounter{
+			m:  make(map[string]int64),
+			mu: &sync.RWMutex{},
+		}
 	}
 
 }
 
-func setBlobberReadCtr(blobber *blockchain.StorageNode, ctr int64) {
+func setBlobberReadCtr(clientID, blobberID string, ctr int64) {
+	key := clientID + blobberID
 	brc.mu.Lock()
-	brc.m[blobber.ID] = ctr
+	brc.m[key] = ctr
 	brc.mu.Unlock()
 }
 
-func getBlobberReadCtr(blobber *blockchain.StorageNode) int64 {
+func getBlobberReadCtr(clientID, blobberID string) int64 {
+	key := clientID + blobberID
```

Comment:
Perhaps with ":" separator?

Reply:
Yes but its redundant. ":" would be good if it was for display purpose.
Why add a byte if without it, it works fine.